src/ncp/ncp_base.cpp

```diff
@@ -5129,6 +5283,46 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SRC_MATCH_EXTENDED_ADDRESSES(uint8_t
 
 #endif
 
+ThreadError NcpBase::SetPropertyHandler_NET_PSKC(uint8_t header, spinel_prop_key_t key, const uint8_t *value_ptr,
+                                                    uint16_t value_len)
+{
+    const uint8_t *ptr = NULL;
+    spinel_size_t len;
+    spinel_ssize_t parsedLength;
+    ThreadError errorCode = kThreadError_None;
+
+    parsedLength = spinel_datatype_unpack(
+                       value_ptr,
+                       value_len,
+                       SPINEL_DATATYPE_DATA_S,
+                       &ptr,
+                       &len
+                   );
+
+    if ((parsedLength > 0) && (len == sizeof(spinel_thread_pskc_t)))
+    {
+        otOperationalDataset dataset;
+        otDatasetGetActive(mInstance, &dataset);
+        uint64_t activeTimestamp = dataset.mActiveTimestamp;
+        memset(&dataset, 0, sizeof(dataset));
+        memcpy(dataset.mPSKc.m8, ptr, sizeof(spinel_thread_pskc_t));
+        dataset.mIsPSKcSet = true;
+        dataset.mActiveTimestamp = activeTimestamp;
+        dataset.mIsActiveTimestampSet = true;
+
+        SuccessOrExit(errorCode = otDatasetSetActive(mInstance, &dataset));
```

Comment:
OK, since the implementation for setting PSKc should be the same as other network parameter such as PANID. I'm going to submit a separate PR for that. This PR will focus on border agent proxy feature only.